lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -1266,7 +1309,21 @@ def _process_logging_relation_changed(self, relation: Relation):
 
     def _endpoints(self) -> List[dict]:
         """Return a list of Loki Push Api endpoints."""
-        return [{"url": self._url(unit_number=i)} for i in range(self._charm.app.planned_units())]
+        endpoints = []
+        path = "/loki/api/v1/push"
+
+        if self._charm.ingress_per_unit.urls:
+            return [{"url": url + path} for url in self._charm.ingress_per_unit.urls.values()]
```

Comment:
Concatenating with something more solid than string concatenation would make me feel safer, e.g., to be sure to avoid double slashes

Reply:
`self._charm.ingress_per_unit.urls` always returns an url without trailing slash, and `path` starts with `/`. Anyway I can add a `rstrip` to the url.